reconcile/utils/slack_api.py

```diff
@@ -1,6 +1,7 @@
+from __future__ import annotations
 import json
 import logging
-from typing import Sequence, Dict, Any, Mapping, Optional, Union
+from typing import Iterable, Protocol, Sequence, Any, Mapping, Optional, Union
```

Comment:
nit: importing this from `typing` is deprecated since 3.9. Those should be imported from `collections.abc`. See https://docs.python.org/3.9/library/typing.html#typing.Iterable as an example